spatialos-sdk/src/worker/connection.rs

```diff
@@ -167,6 +168,55 @@ pub struct WorkerConnection {
     attributes: Vec<String>,
 }
 
+#[allow(unused_assignments)]
+fn do_connect<F>(connect: F, params: &ConnectionParameters) -> WorkerConnectionFuture
+where
+    F: FnOnce(&Worker_ConnectionParameters) -> WorkerConnectionFuture,
+{
+    let mut conn_params = params.to_worker_sdk();
+
+    // conn_params.network will not have the protocol parameters setup (due to requiring pointers to things on the stack). Therefore, we do that here.
+    let mut udp_kcp_params = Worker_Alpha_KcpParameters::default();
+    let mut udp_erasure_codec_params = Worker_ErasureCodecParameters::default();
+    let mut udp_heartbeat_params = Worker_HeartbeatParameters::default();
+    let mut udp_flow_control_params = Worker_Alpha_FlowControlParameters::default();
+    match &params.network.protocol {
+        ProtocolType::Tcp(params) => {
```

Comment:
Small nit - can we not alias `params` from the function signature here? I was really confused when reading the match block

Reply:
Sure